danar/src/main/java/info/nightscout/androidaps/danaRv2/comm/MsgHistoryEvents_v2.kt

```diff
@@ -81,17 +83,26 @@ class MsgHistoryEvents_v2 constructor(
                     pumpId = datetime,
                     pumpType = PumpType.DANA_RV2,
                     pumpSerial = danaPump.serialNumber)
+                aapsLogger.debug(LTag.PUMPBTCOMM, (if (newRecord) "**NEW** " else "") + "EVENT TEMPSTART (" + recordCode + ") " + dateUtil.dateAndTimeString(datetime) + " (" + datetime + ")" + " Ratio: " + param1 + "% Duration: " + param2 + "min")
                 status = "TEMPSTART " + dateUtil.timeString(datetime)
+                if (newRecord)
```

Comment:
These events are not user events. these are result of AAPS work and should not be looged as "manual" actions
I didn't check other places but we should only log actions when user click something

Reply:
Here I thought it was manual entries that the user entered directly on pump (and imported in AAPS from pump history), that's why I proposed to add them.

For all other uel.log, most of them are "click" done by user in AAPS.

The exceptions are (and I can remove all or some of them if you want):
- in LoopPlugin (uel.log for SMB and TBR) (these uel.log are recorded but hidden by default in UserEntryFragment)
- in Automation Plugin (uel.log for TBR, PS, Loop Enabled/Disabled done by an automation rule, indirect actions done by the user but not with a click)
- in NSClient Plugin (uel.log for new entries done from NsClient Plugin, equivalent of uel.log of SMS Plugin but for NSClient entries)
- Dexcom Plugin (Sensor change or Calibration done by user in Dexcom Patched app)

I thought that in case of investigation, all these entries could be usefull in the csv file...

Reply:
the rest seems to be ok except NsClientAddUpdateWorker which will generate a lot of events until we finish migration to API
so, for now remove only these Dana* history
we can review the rest when we finish all changes

Reply:
Gotta say I don't really like the idea of having `UserEntry` code at the driver level.
For boluses and TBRs, it could be reported back whether a bolus was "found" on the pump, or read after AAPS requested it, so that `PumpSync` (or the service calling it) could do such logging, or record the origin of it.
Entirely related to that: the Combo produces notifications and dispatches them via `rxBus`. Which makes me wonder whether including methods in `PumpSync` to report events back to AAPS might be a good idea.
That way the driver could focus on driving the pump and wouldn't need to access `uel`, `rxBus` or `Notification` ...